SPHINXsys/src/shared/simbody_sphinxsys/xml_engine.h

```diff
@@ -67,22 +67,26 @@ namespace SPH
 		/**Add child element to a given element. */
 		void addChildToElement(SimTK::Xml::Element &father_element, const std::string &child_name);
 
-		/** Add an attribute of type string to an xml element.  */
-		void setAttributeToElement(const SimTK::Xml::element_iterator &ele_ite, const std::string &attrib_name, const Real &value)
-		{
-			SimTK::Xml::Attribute attr_(attrib_name, SimTK::String(value));
-			ele_ite->setAttributeValue(attr_.getName(), attr_.getValue());
-		};
+		//----------------------------------------------------------------------
+		//	Add an attribute of type string to an xml element.
+		//----------------------------------------------------------------------
+		void setAttributeToElement(const SimTK::Xml::element_iterator &ele_ite, const std::string &attrib_name, const Real &value);
+		void setAttributeToElement(const SimTK::Xml::element_iterator& ele_ite, const std::string& attrib_name, const int& value);
+
 		template <int DIMENSION>
-		void setAttributeToElement(const SimTK::Xml::element_iterator &ele_ite, const std::string &attrib_name,
-								   const Eigen::Matrix<Real, DIMENSION, 1> &value)
+		void setAttributeToVectorElement(const SimTK::Xml::element_iterator& ele_ite, const std::string& attrib_name, 
+										 const Eigen::Matrix<Real, DIMENSION, 1>& value)
 		{
 			SimTK::Xml::Attribute attr_(attrib_name, SimTK::String(EigenToSimTK(value)));
 			ele_ite->setAttributeValue(attr_.getName(), attr_.getValue());
 		};
+
+		void setAttributeToElement(const SimTK::Xml::element_iterator &ele_ite, const std::string &attrib_name, const Vec2d &value);
+		void setAttributeToElement(const SimTK::Xml::element_iterator& ele_ite, const std::string& attrib_name, const Vec3d& value);
+
 		template <int DIMENSION>
-		void setAttributeToElement(const SimTK::Xml::element_iterator &ele_ite, const std::string &attrib_name,
-								   const Eigen::Matrix<Real, DIMENSION, DIMENSION> &value)
+		void setAttributeToMatrixElement(const SimTK::Xml::element_iterator& ele_ite, const std::string& attrib_name,
+										 const Eigen::Matrix<Real, DIMENSION, DIMENSION>& value)
 		{
 			SimTK::Array_<Real, int> array_(DIMENSION * DIMENSION);
 			for (int i = 0; i < DIMENSION; i++)
```

Comment:
Why is this done only for matrices and not vectors ?

Both should be serializable uniformly according to [SimTK::String(const T&)](https://simtk.org/api_docs/simbody/api_docs33/Simbody/html/classSimTK_1_1String.html#ae1c3122a261b2a809607040facdd7c07) and [operator<< for SimTK::Mat](https://simtk.org/api_docs/simbody/api_docs33/Simbody/html/namespaceSimTK.html#abf8a923ce5ce85ac3fddbddd1c5df70d). This would reduce duplication and simplify the code to do the same for vectors and matrices.

Reply:
Another reason the String function is for simtk and our data is from eigen3. 
This leads to some issue.
However, any better revision is welcomed.